mozetl/taar/taar_lite_guidguid.py

```diff
@@ -0,0 +1,189 @@
+AMO_DUMP_KEY = 'telemetry-ml/addon_recommender/addons_database.json'
+MAIN_SUMMARY_PATH = 's3://telemetry-parquet/main_summary/v4/'
+ONE_WEEK_AGO = (dt.datetime.now() - dt.timedelta(days=7)).strftime('%Y%m%d')
+
+
+def extract_telemetry(spark):
+    """ load some training data from telemetry given a sparkContext
+    """
+    sc = spark.sparkContext
+
+    # Define the set of feature names to be used in the donor computations.
+
+    def get_initial_sample(pct_sample=10):
+        # noqa: ignore=E127,E502,E999
+        """ Takes an initial sample from the longitudinal dataset
+        (randomly sampled from main summary). Coarse filtering on:
+        - number of installed addons (greater than 1)
+        - corrupt and generally wierd telemetry entries
+        - isolating release channel
+        - column selection
+        -
+        - pct_sample is an integer [1, 100] indicating sample size
+        """
+        # Could scale this up to grab more than what is in longitudinal and see how long it takes to run.
+        return (spark.table("longitudinal")
+                .where("active_addons IS NOT null")
+                .where("size(active_addons[0]) > 1")
+                .where("normalized_channel = 'release'")
+                .where("build IS NOT NULL AND build[0].application_name = 'Firefox'")
+                .selectExpr("client_id as client_id", "active_addons[0] as active_addons"))
+
+    def get_addons_per_client(users_df):
+        """ Extracts a DataFrame that contains one row
+        for each client along with the list of active add-on GUIDs.
+        """
+
+        def is_valid_addon(guid, addon):
+            """ Filter individual addons out to exclude, system addons,
+            legacy addons, disabled addons, sideloaded addons.
+            """
+            return not (
+                addon.is_system or
+                addon.app_disabled or
+                addon.type != "extension" or
+                addon.user_disabled or
+                addon.foreign_install or
+                # make sure the amo_whitelist has been broadcast to worker nodes.
+                guid not in broadcast_amo_whitelist.value
+            )
+
+        # TODO: may need additional whitelisting to remove shield addons
+        # it should be populated from the current list of shield studies.
+
+        # Create an add-ons dataset un-nesting the add-on map from each
+        # user to a list of add-on GUIDs. Also filter undesired add-ons.
+        return (
+            users_df.rdd
+                    .map(lambda p: (p["client_id"],
+                                    [guid for guid, data in p["active_addons"].items() if is_valid_addon(guid, data)]))
+                    .filter(lambda p: len(p[1]) > 1)
+                    .toDF(["client_id", "addon_ids"])
+        )
+
+    logging.info("Init loading client features")
+    client_features_frame = get_initial_sample()
+    logging.info("Loaded client features")
```

Comment:
Spark [lazily evaluates](https://spark.apache.org/docs/2.2.0/rdd-programming-guide.html#rdd-operations) the RDDs/DataFrames. This means that the logging information is not actually capturing events as they happen. This limits the usefulness of logging in this context.

Reply:
Ah yeah, nice catch, I forgot that `toDF()` doesn't force materialization of the DF.
These lines can be removed to minimize noise in the application log.

Reply:
Dropped the log messages.